src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java

```diff
@@ -58,19 +58,30 @@ public SuccessfulTransformationResponse transformExceptionRecord(
             );
         } catch (HttpStatusCodeException ex) {
             log.error(
-                "Failed to transform Exception Record to case data for case type {} ",
+                "Failed to transform Exception Record to case data for case type {} and id {}",
                 exceptionRecord.caseTypeId,
+                exceptionRecord.id,
                 ex
             );
 
             if (ex.getStatusCode().equals(UNPROCESSABLE_ENTITY) || ex.getStatusCode().equals(BAD_REQUEST)) {
-                throw new InvalidCaseDataException(
-                    ex,
-                    objectMapper.readValue(ex.getResponseBodyAsByteArray(), TransformationErrorResponse.class)
-                );
+                tryParseResponseBody(ex);
             }
 
             throw new CaseTransformationException(ex, ex.getResponseBodyAsString());
         }
     }
+
+    private void tryParseResponseBody(HttpStatusCodeException exception) throws CaseTransformationException {
+        try {
+            TransformationErrorResponse errorResponse = objectMapper.readValue(
+                exception.getResponseBodyAsByteArray(),
+                TransformationErrorResponse.class
+            );
+
+            throw new InvalidCaseDataException(exception, errorResponse);
```

Comment:
do you need this ?

Reply:
as per specification, we should accept custom response object defined in there. applies for 400s and 422s. this is as before just "safer" - catch deserialisation issues and wrap with custom exception instead of forwarding it to the source